clarity_ext/repository/step_repository.py

```diff
@@ -131,6 +131,9 @@ def update_objects(self, objects):
             if type(obj) == Analyte:
                 original_analyte_from_rest = ApiArtifact(self.session.api, id=obj.id)
                 update_queue.append(obj.updated_rest_resource(original_analyte_from_rest, self.udf_map))
+            elif type(obj) == ResultFile:
```

Comment:
I generally prefer to throw NotImplemented when something isn't implemented, and instead change the tests/code that's using this to not do that until it's ready (fail-fast). Is there some reason that's not possible in this case?